internal/services/datafactory/data_factory_linked_service_azure_function_resource.go

```diff
@@ -211,6 +239,14 @@ func resourceDataFactoryLinkedServiceAzureFunctionRead(d *pluginsdk.ResourceData
 	d.Set("additional_properties", azureFunction.AdditionalProperties)
 	d.Set("description", azureFunction.Description)
 
+	if functionKey := azureFunction.AzureFunctionLinkedServiceTypeProperties.FunctionKey; functionKey != nil {
+		if keyVaultKey, ok := functionKey.AsAzureKeyVaultSecretReference(); ok {
```

Comment:
We should probably return an error if this fails (similarly to https://github.com/hashicorp/terraform-provider-azurerm/pull/13159/files#diff-c0912934de2e79bc15b51ee8686117dc210319bf9aa99c439b627ee354512a78R233-R235)